toolkit/tools/pkg/imagecustomizerlib/customizepackages_test.go

```diff
@@ -211,3 +215,37 @@ func TestCustomizeImagePackagesDiskSpace(t *testing.T) {
 	assert.ErrorContains(t, err, "failed to customize raw image")
 	assert.ErrorContains(t, err, "failed to install package (gcc)")
 }
+
+func ensureTdnfCacheCleanup(t *testing.T, imageConnection *ImageConnection, dirPath string) error {
+	// Array to capture all the files of the provided root directory
+	var existingFiles []string
+
+	// Start the directory walk from the initial dirPath and collect all existing files
+	fullPath := filepath.Join(imageConnection.Chroot().RootDir(), dirPath)
+	err := filepath.WalkDir(fullPath, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return fmt.Errorf("Failed to access path (%s): %w", path, err)
+		}
+		// Ignore files in the local-repo folder if the base image version is 2.0
+		if !(strings.Contains(path, "local-repo") && baseImageVersionDefault == "2.0") {
+			fileInfo, err := os.Stat(path)
+			if err != nil {
+				return fmt.Errorf("failed to get file info for %s: %w", path, err)
+			}
+			if !fileInfo.IsDir() {
+				// Append the file to the existingFiles array
+				existingFiles = append(existingFiles, path)
+			}
+		}
+		return nil
+	})
+
+	if err != nil {
```

Comment:
Use `assert.NoError(t, err)`

The caller of `ensureTdnfCacheCleanup` is currently ignoring the returned error.